aguda.akinwunmi-AkinAguda/main.js

```diff
@@ -0,0 +1,156 @@
+            maxWidth = maxWidth;
+        }
+        else{
+            shouldStartSlide = false;
+            maxWidth = 100;
+            unit = `%`;
+        }
+        fetch(section)
+        .then(function(result){
+            return(result.json());
+        })
+        .then(function(print){
+            var length = print.articles.length;
+            var each = [0];
+            for (i = each.length; i < length; i++){
+                each.push(i);
+            }
+            if(random){
+                for(i = each.length-1; i > 0; i--){
+                    var randomNo = Math.floor(Math.random()*(i+1));
+                    lasteach = each[i];
+                    each[i] = each[randomNo];
+                    each[randomNo] = lasteach;
+                }
+            }
+            console.log(each)
+            console.log(print)
+            for(i = 0; i != number; i++){
+                if (tileOrBlock == "block"){
+                    var cont = document.getElementById(id);
+                    cont.setAttribute("style", `width:${maxWidth}${unit};`)
+                    cont.innerHTML += `<div class = "card" style = min-width:${maxWidth}${unit} id = "id-${id}";>
+                    <img src="${print.articles[each[i]].urlToImage}" 
+                    class="news-image"><p class = "caption">${print.articles[each[i]].title}"</p>
+                    <p class="story">${print.articles[each[i]].description}</p>
+                    <p class = "date">${print.articles[each[i]].source.name} on ${print.articles[each[i]].publishedAt}</p></div>`;
+                }
+                if (tileOrBlock == "tile"){
+                    var cont = document.getElementById(id);
+                    cont.setAttribute("style", `max-width:${maxWidth}${unit};`)
+                    cont.innerHTML += `<div class = "tile" style = min-width:${maxWidth}${unit};>
+                    <div>
+                    <img src="${print.articles[each[i]].urlToImage}" 
+                    class="tileNews-image">
+                    </div>
+                    <div>
+                    <p class = "tileCaption">${print.articles[each[i]].title}"</p>
+                    <p class = "tileDate">${print.articles[each[i]].source.name} on ${print.articles[each[i]].publishedAt}
+                    </p></div>`;
+                }
+            }
+            if(shouldStartSlide){
+                startSlide = true;
+                setInterval(move(maxWidth), 5000);
+            }
+        })
+    }
+}
+var techCrunch = "https://newsapi.org/v2/top-headlines?sources=techcrunch&apiKey=78c7a959be2d49cb8b88e9a2895ed5c9" ;
+var topUs = "https://newsapi.org/v2/top-headlines?country=us&category=business&apiKey=78c7a959be2d49cb8b88e9a2895ed5c9";
+var news = new getNews("req", techCrunch, 5, false, "block", 400);
+var tiledNews = new getNews("tiles", techCrunch, 4, false, "tile", 400);
+var tiledTopNews = new getNews("top-stories", topUs, 5, false, "tile", 400);
```

Comment:
Alternatively you can do
```js
var getNews = function(id, section, number, random, tileOrBlock, maxWidth) {
    return {
      fetchApi = () => {
        var unit = `px`;
         ...
      }
    }
}
news = getNews("req", techCrunch, 5, false, "block", 400);
news.fetchApi();
```
We have only one function call, no need for creating more with `new`.  
It's advisable not to use OOP in ES5 except you really know what you're doing

Reply:
Yeah. Thanks a lot for this. Was thinking about it too